src/muya/lib/selection/index.js

```diff
@@ -619,9 +619,13 @@ class Selection {
         // range.collapse(true)
         let rects = range.getClientRects()
         if (rects.length === 0) {
-          rects = range.startContainer && range.startContainer.nodeType === Node.ELEMENT_NODE
-            ? range.startContainer.getClientRects()
+          rects = range.startContainer ? range.startContainer.parentElement.getClientRects()
```

Comment:
@MrHeer Thanks, the fix working fine and I hope you could add a check for `ELEMENT_NODE` and `TEXT_NODE` like:

```js
          rects = []
          if (range.startContainer && (range.startContainer.nodeType === Node.ELEMENT_NODE || range.startContainer.nodeType === Node.TEXT_NODE)) {
            rects = range.startContainer.parentElement.getClientRects()
          }

          // Prevent tiny vibrations
```

Reply:
@fxha I had updated. Can you test it?